appveyor.yml

```diff
@@ -115,101 +91,47 @@ install:
-  # Therefore, we install swig here in order to create the swig.exe shim file.
-  # TODO: Remove the circumvention once it works in pywbem_os_setup.bat.
-  #- choco install -y --allow-empty-checksums swig
-  #- where swig
-  #- swig -version
+  - if "%UNIX_PATH%"=="none" where make
+  - if "%UNIX_PATH%"=="none" make --version
 
   # Install wget. Used by make to install pywbem scripts
   - if "%UNIX_PATH%"=="none" tools\retry choco install -y wget
   - if "%UNIX_PATH%"=="none" where wget
   - if "%UNIX_PATH%"=="none" wget --version
 
-  # Examine the final environment
-  - 'echo: %PATH%'
-  - 'echo: %PYTHONPATH%'
-  - 'echo: %INCLUDE%'
-  - 'echo: %LIB%'
-  - 'echo: %SWIG_FEATURES%'
-
-  # Verify that the commands used in pywbem_os_setup.bat are available
-  # Already verified further up: choco, pip, python.
-  - where chmod
-  - chmod --version
-  - where tar
-  - tar --version
-
-  # Install Python virtualenv and OS commands
-  #
-  # Note: Installing Swig in pywbem_os_setup.bat does not work when it runs
-  # inside of make which runs inside of tox, because it does not generate the
-  # swig.exe shim file (using GenShim). See GenShim issue
-  # https://github.com/chocolatey/shimgen/issues/43.
-  # Therefore, we invoke pywbem_os_setup.bat outside of make.
-  - pip install virtualenv
-  - virtualenv myenv
-  - myenv\Scripts\activate
-  - where python
-  - python --version
-  - where pip
-  - pip --version
-
-  # Install Python packages and OS-level prerequisites
-  - make platform env
-  - pip list
-  - make install
-  - pip list
-  - make develop
-  - pip list
-
-# Not a C# project, build stuff at the test step instead.
+  # Install Pip, where needed
+  - if "%UNIX_PATH%"=="C:\cygwin\bin" ( bash --noprofile --norc -c "set -e; %PYTHON_CMD% -m ensurepip; %PYTHON_CMD% -m pip install --upgrade pip setuptools wheel" )
+  - if "%UNIX_PATH%"=="C:\cygwin64\bin" ( bash --noprofile --norc -c "set -e; %PYTHON_CMD% -m ensurepip; %PYTHON_CMD% -m pip install --upgrade pip setuptools wheel" )
+
+  # Install Tox
+  - if "%UNIX_PATH%"=="none" ( cmd /c "pip install tox & where tox & tox --version" )
+  - if "%UNIX_PATH%"=="C:\cygwin\bin" ( bash --noprofile --norc -c "set -e; %PYTHON_CMD% -m pip install tox; which -a tox; tox --version" )
+  - if "%UNIX_PATH%"=="C:\cygwin64\bin" ( bash --noprofile --norc -c "set -e; %PYTHON_CMD% -m pip install tox; which -a tox; tox --version" )
+
+  # Initialize CygWin updates (must be done outside of CygWin bash in case it updates cygwin.dll)
+  - 'if "%UNIX_PATH%"=="C:\cygwin\bin" ( C:\cygwin\setup-x86.exe --no-shortcuts --quiet-mode )'
+  - 'if "%UNIX_PATH%"=="C:\cygwin64\bin" ( C:\cygwin64\setup-x86_64.exe --no-shortcuts --quiet-mode )'
+
 build: false
 
 before_test:
+  - 'if "%UNIX_PATH%"=="C:\cygwin\bin" (bash export & export LC_ALL=C.UTF-8 & export LANG=C.UTF-8)'
+  - 'if "%UNIX_PATH%"=="C:\cygwin64\bin" (bash export & export LC_ALL=C.UTF-8 & export LANG=C.UTF-8)'
```

Comment:
This line fails in the Appveyor run with cygwin64. What do you want to do there? The environment variable that gets set, disappears again when the bash process ends...

Reply:
It was what is called "Don't do anything when you are tired".

Reply:
However, the actual issue is that with python3 click has issues if a unicode local is not set and that is apparently the case with cygwin.  So I am setting the lcoal.  Fixed my really stupid code and it now works.  See https://click.palletsprojects.com/en/7.x/python3/